src/thirdParty/registerMixin.js

```diff
@@ -1,25 +1,33 @@
+import { lib } from '../lib.js';
 import mixins from '../mixins/index.js';
 
 /**
- * Register a mixin.
+ * Register an mixin to cornerstoneTools.
  *
- * @param  {Object} newMixin The mixin to register.
  * @param {string} name The name of the mixin.
+ * @param {Object} mixin The mixin itself.
+ * @param {boolean} [overwrite=false] Whether an mixin should be overwritten,
+ *                                    should it have the same name.
  */
-export default function (newMixin, name) {
+export default function (name, mixin, overwrite = false) {
```

Comment:
If the logic for `registerX` is similar, you can make a single function, and then spin off a version per type using `.bind()`?

- Less code
- Changes update each type
- Less to unit test

Reply:
RE: `setToolMode`